elements/pf-background-image/pf-background-image.ts

```diff
@@ -0,0 +1,122 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { queryAssignedElements } from 'lit/decorators/query-assigned-elements.js';
+import { property } from 'lit/decorators/property.js';
+
+import styles from './pf-background-image.css';
+
+/**
+ * A **background image** allows you to place an image in the background of your page or area of a page.
+ * @summary Allows users to place an image in the background of your page or area of a page.
+ *
+ * @slot filter - Overrides the default svg filter for the background image.
+ * @slot - Default content slot.
+ *
+ * @cssprop {<color>} --pf-c-background-image--BackgroundColor {@default `#151515`}
+ * @cssprop --pf-c-background-image--BackgroundImage
+ * @cssprop --pf-c-background-image--BackgroundImage-2x
+ * @cssprop --pf-c-background-image--BackgroundImage--sm
+ * @cssprop --pf-c-background-image--BackgroundImage--sm-2x
+ * @cssprop --pf-c-background-image--BackgroundImage--lg
+ * @cssprop --pf-c-background-image--Filter {@default `url("#filter")`}
+ *
+ */
+@customElement('pf-background-image')
+export class PfBackgroundImage extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * The URL for the image shown on mobile
+   */
+  @property({ reflect: true }) src?: string;
+
+  /**
+   * The image shown on mobile with 2x DPI
+   */
+  @property({ reflect: true, attribute: 'src-2x' }) src2x?: string;
+
+  /**
+   * The URL for the image shown on small screens (min-width: 576px)
+   */
+  @property({ reflect: true, attribute: 'src-sm' }) srcSm?: string;
+
+  /**
+   * The URL for the image shown on small screens (min-width: 576px) with 2x DPI
+   */
+  @property({ reflect: true, attribute: 'src-sm-2x' }) srcSm2x?: string;
+
+  /**
+   * The URL for the image shown on large screens (min-width: 992px)
+   */
+  @property({ reflect: true, attribute: 'src-lg' }) srcLg?: string;
+
+  /**
+   * Apply SVG Filter to the image
+   */
+  @property({ type: Boolean, reflect: true }) filter = false;
+
+  @queryAssignedElements({ slot: 'filter', selector: 'svg' }) private _svg?: SVGElement[];
+
+  #svg?: SVGElement;
+
+  #updated = false;
+
+  render() {
+    return html`
+      <div id="container" part="container" style="${this.#cssVars()}">
```

Comment:
better to do this as a styleMap, and you can keep everything in the render method IMO

Reply:
💯 didn't even think about that